codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -181,8 +183,8 @@ public ChannelFuture writeData(final ChannelHandlerContext ctx, final int stream
         }
 
         // Hand control of the frame to the flow controller.
-        ChannelFuture future =
-                flowController().sendFlowControlledFrame(ctx, stream, data, padding, endOfStream, promise);
+        ChannelFuture future = flowController().sendFlowControlledData(ctx, stream,
+                new FlowControlledData(ctx, streamId, data, padding, endOfStream, promise), promise);
```

Comment:
Ah ok, I didn't think I fully appreciated what you were trying to do here :).  That's a nice improvement!  Let's just move the new flow controller writing interface to a sub-interface of the `RemoteFlowController`, so that it's purpose is more clear.  Sound good?